src/main/haskell/kore/test/Test/Kore/Builtin/Set.hs

```diff
@@ -174,6 +176,89 @@ asSymbolicPattern result =
     applyElement key = App_ symbolElement [key]
     applyConcat set1 set2 = App_ symbolConcat [set1, set2]
 
+{- | Check that unifying a concrete set with itself results in the same set
```

Comment:
These tests look good. Please add some integration tests in `Test.Kore.Step.Simplification.Integration` like `test_substituteMap`.